pkg/kubectl/resource/builder.go

```diff
@@ -274,12 +275,47 @@ func (b *Builder) SelectAllParam(selectAll bool) *Builder {
 	return b
 }
 
-// ResourceTypeOrNameArgs indicates that the builder should accept arguments
-// of the form `(<type1>[,<type2>,...]|<type> <name1>[,<name2>,...])`. When one argument is
-// received, the types provided will be retrieved from the server (and be comma delimited).
+func hasNamesArg(args []string) bool {
```

Comment:
Please add a unit test for this function-- seeing the cases laid out in a table will be very helpful for understanding it.